src/common/Throttle.h

```diff
@@ -338,17 +338,29 @@ class TokenBucketThrottle {
   struct Bucket {
     CephContext *cct;
     const std::string name;
-    std::atomic<uint64_t> remain = { 0 }, max = { 0 };
 
-    Bucket(CephContext *cct, const std::string& n, uint64_t m)
-      : cct(cct), name(n),
-	remain(m), max(m)
-    {
-    }
+    std::atomic<uint64_t> cir = { 0 };
+    std::atomic<uint64_t> cbs = { 0 };
+    std::atomic<uint64_t> remain = { 0 };
+
+    std::atomic<uint64_t> tick = { 50 };
+    std::atomic<uint64_t> ticks_per_second = { 0 };
+    std::atomic<uint64_t> tokens_per_tick = { 0 };
+    std::atomic<uint64_t> _divide_left = { 0 };
```

Comment:
what does cir, cbs and _devide_left mean? and are they atomic?

Reply:
CIR (Committed Information Rate) is the rate that tokens generated.
CBS (Committed Burst Size) is the size of the TokenBucket, which is also means the burst limit.

Reply:
... also agree about the ```atomic``` question. Shouldn't these be protected already by the lock?